docs/standard/commandline/syntax.md

```diff
@@ -0,0 +1,577 @@
+| 1   | *n*     | Valid:           | --file a.json               |
+|     |         | Valid:           | --file a.json b.json        |
+|     |         | Invalid:         | --file                      |
+
+### Option overrides
+
+If the arity maximum is 1, `System.CommandLine` can still be configured to accept multiple instances of an option. In that case, the last instance of a repeated option overwrites any earlier instances. In the following example, the value 2 would be passed to the `myapp` command.
+
+```console
+myapp --delay 3 --message example --delay 2
+```
+
+### Multiple arguments
+
+`System.CommandLine` can be configured to accept multiple arguments for one option without repeating the option name.
+
+In the following example, the list passed to the `myapp` command would contain "a", "b", "c", and "d":
+
+```console
+myapp --list a b c --list d
+```
+
+This syntax pattern isn't enabled for .NET CLI commands.
+
+## Option bundling
+
+POSIX recommends that you support *bundling* of single-character options, also known as *stacking*. Bundled options are single-character option aliases specified together after a single hyphen prefix. The options before the last one in a bundle can't have required arguments. For example, the following command lines are equivalent:
+
+```console
+git clean -f -d -x
+git clean -fdx
+```
+
+If an argument is provided after an option bundle, it applies to the last option in the bundle. The following command lines are equivalent:
+
+```console
+myapp -a -b -c arg
+myapp -abc arg
+```
+
+In both variants in this example, the argument `arg` would apply only to the option `-c`.
+
+## Boolean options (flags)
+
+If `true` or `false` is passed for an option having a `bool` argument, it's parsed as expected. But an option whose argument type is `bool` typically doesn't require an argument to be specified. Boolean options, sometimes called "flags", typically have an [arity](#argument-arity) of `ArgumentArity.ZeroOrOne`. The presence of the option name on the command line, with no argument following it, results in a default value of `true`. The absence of the option name in command-line input results in a value of `false`. If the `myapp` command prints out the value of a Boolean option named `--interactive`, the following input creates the following output:
+
+```console
+myapp
+myapp --interactive
+myapp --interactive false
+myapp --interactive true
+```
+
+```output
+False
+True
+False
+True
+```
+
+In the .NET CLI, this pattern doesn't always hold true. Some Boolean options result in the same behavior when you pass `false` as when you pass `true`. This behavior results when .NET CLI code that implements the option only checks for the presence or absence of the option, ignoring the value. An example is `--no-restore` for the `dotnet build` command. Pass `no-restore false` and the restore operation will be skipped the same as when you specify `no-restore true` or `no-restore`.
```

Comment:
My opinion is solidfiying. Somewhere in our world we should have "How the .NET CLI is weird". It isn't that important when people are learning SCL, and it might be to someone that isn't ever going to use SCL